### ISSUE TITLE
Some questions about implementation and training

### ISSUE THREAD

**Issue by gogolgrind** [layumi/2015_Face_Detection#12]
Hello. for how i work on rapid object detection problem. And I decided to use  2015 CVPR Cascade CNNs for Face Detection as base solution. I am developing it with theano\lasagne\python.
So now I have some questions.
- It is seems , that you use different architectures for  networks( more conv layers, and one drop layer before output. Is it heuristics approach?
- How do you train detection nets? More concretely, witch patches do you use as negative examples? and how many examples do you use? 
-  Similar question about calibration net. How to choose distribution of 45 calibration classes?
- How do  you implemented 24 net multi output? (see picture)
  ![cnn](http://cs630417.vk.me/v630417831/3832d/kcNHSxz8530.jpg)
  If you want you can write answers for me by email k.sozykin@innopolis.ru or just here 
  Thank you and best wishes !

**Comment by layumi** [comment]
@gogolgrind 
1. The net which is deeper has been proved to a better performance(Like res-net). And dropout can help net to be more robust and perform well on validation.
2. I find background images on imagine (buildings,planets and so on).  About 200,000 pictures is 4~5 times of positive data(about 50,000 after mirror)
3. For example, if label is to enlarge the bounding box, you can take a small bounding box as  input.
You can reverse the method to produce the data.
4. I write a layer to concat 12net fc layer to 24net fc layer. You can refer to vl_simple.m custom layer and custom48 layer.

**Comment by gogolgrind** [comment]
Hello. I try to convert you pre-trained model from test brach for use it in my theano implementation.
Unfortunately, there are some issues. For example there is no softmax layer in f12net.mat. More over structure of this fie different from definition in matlab code.  Can you provide for me  correct pre - trained model?

**Comment by layumi** [comment]
@gogolgrind 
1.Actually I wrote softmax layer in my test code. You can see something like "T=exp(X-max(X));Y=X/T;" in my test code.
2.You can refer to the model file and find its structure. The definition is quite out-of-date.
